delphi-frontend/src/main/antlr3/au/com/integradev/delphi/antlr/Delphi.g

```diff
@@ -237,13 +250,14 @@ import org.apache.commons.lang3.StringUtils;
           break;
 
         case EOF:
-          throw new LexerException(
+          throwLexerException(
               "line "
                   + state.tokenStartLine
                   + ":"
                   + state.tokenStartCharPositionInLine
                   + " unterminated multi-line comment",
               state.tokenStartLine);
+          break;
```

Comment:
Is `state.tokenStartLine` and `state.tokenStartCharPositionInLine` relative to the current file here? This might still cause file pointer issues if there are lexer exceptions in an include file.

Reply:
Or is this made okay by the new `DelphiIncludeLexer`?

Reply:
Those are relative to the current file but they only appear in the exception message.
The `DelphiIncludeLexer` overrides `throwLexerException` to add additional information to the message and also passes the correct line number through.

Reply:
On second glance at `DelphiIncludeLexer::throwLexerException`, I think I can actually simplify the override a little bit due to the recursive `DelphiToken` include position calculation I mentioned earlier.
I'll try that now.

Reply:
Simplified it.